### ISSUE TITLE
Controling signs of context variables in Prophet

### ISSUE THREAD

**Issue by Angylina** [facebookexperimental/Robyn#635]
## Describe issue

Hi everyone, 
I have a question about controlling the sign of context variables. We do have context_signs, which control it in the model itself. However, when a context_variable is also a factor variable, prophet uses them as additional information. Then, in the prophet decomposition we can see how the effect from this variable looks. 

## Provide reproducible example

I have two factor variables, which are simply 1 in case of event, 0 when there is no event. For factor 1 I expect to have a positive effect on revenue, while for factor 2 I expect a negative effect. Inside robyn_inputs I specify:
context_vars = c("other_variable","factor_1",'factor_2'),
  context_signs=c('positive','positive','negative'),

This works well in the model itself - in the waterfall chart I have the contribution of factor 2 as negative.

However, this is what the Prophet decomposition for factor 2 looks like - despite specifying the context_sign as negative, Prophet attributes positive effect to it. 

![Schermopname (77)](https://user-images.githubusercontent.com/120499663/221914131-ed8e0e5f-1292-461a-9057-491c8cebe2e3.png)c

My question is: is there a way to specify the sign, so that Prophet would always estimate the effect of a variable as negative? Also, do I understand correctly that these context variables are used in both Robyn and Prophet as predictors? 

## Environment & Robyn version
Make sure you're using the latest Robyn version before you post an issue.
- Robyn 3.9.0.
- R version 4.2.2

**Comment by gufengzhou** [comment]
Check ?robyn_inputs. You can specify prophet_signs to be negative. No context vars are not used in prophet as predictors.

**Comment by Angylina** [comment]
I tried using these, but these only work for trend, holiday and season. If I try specifying my factor variable in there, it gives: 

 'prophet_signs' must have same length as 'prophet_vars'

And in prophet_vars I can't specify anything else, except for holiday, season etc. 

Am I missing something or does it indeed not work for factor variables?

**Comment by gufengzhou** [comment]
Ok I got it. The beta sign for your context_vars from the model is more important, right? Do you get negative for factor 2 in the final decomp?

**Comment by Angylina** [comment]
If I specify context variables sign, then the sign is what I need it to be in the final decomp. I specify negative sign for factor 2, then in the model it is negative, but still shows as positive in prophet decomposition. I wanted to get rid of that inconsistency, but if it is not influential then I can leave it. 

As far as I understand, all we get from prophet is trend + seasonality + holidays as predictors, but factor variables inside prophet just help with getting those values from the dependent variable. Is that correct? 

In other words, the question I'm asking is: how influential is the wrong sign for a factor variable in the prophet decomposition, considering that this variables has a correct sign in the model itself?

**Comment by gufengzhou** [comment]
You can ignore the sign from prophet decomp
